man/systemd-resolved.service.xml

```diff
@@ -132,16 +143,43 @@
     lookup zones on all matching interfaces). If the lookup failed on
     all interfaces, the last failing response is returned.</para>
 
-    <para>Routing of lookups may be influenced by configuring
-    per-interface domain names. See
-    <citerefentry><refentrytitle>systemd.network</refentrytitle><manvolnum>5</manvolnum></citerefentry>
-    for details. Lookups for a hostname ending in one of the
-    per-interface domains are exclusively routed to the matching
-    interfaces.</para>
+    <para>Routing of lookups may be influenced by configuring per-interface domain names and other settings. See
+    <citerefentry><refentrytitle>systemd.network</refentrytitle><manvolnum>5</manvolnum></citerefentry> and
+    <citerefentry><refentrytitle>resolvectl</refentrytitle><manvolnum>1</manvolnum></citerefentry> for details. The
+    following query routing logic applies for unicast DNS traffic:</para>
+
+    <itemizedlist>
+      <listitem><para>If a name to look up matches any of the configured search or route-only domains of any link (or
+      the globally configured DNS settings), the "best matching" search/route-only domain is determined: the matching
+      one with the most labels. The query is then sent to all DNS servers of any links or the globally configured DNS
```

Comment:
"If no answer, the query is then sent.."

Reply:
oh, this is actually not what is meant here. We first determine the "best matching" domain, and then look for all links that have this "best matching" domain among their configured search domains. The query is then sent to all of them in parallel. This might be a single link or multiple, if the admin specified the same domains on two interfaces.

I will reword this in the man page